download/download_test.go

```diff
@@ -518,6 +520,109 @@ func TestTriggerManualWithTimeout(t *testing.T) {
+	config.Polling.LongPollingTimeoutSeconds = &timeout
+
+	if err := config.ValidateAndInjectDefaults(); err != nil {
+		t.Fatal(err)
+	}
+
+	fixture := newTestFixture(t)
+	fixture.d = New(config, fixture.client, "/bundles/test/bundle1").WithCallback(fixture.oneShot)
+	fixture.server.longPoll = true
+	fixture.server.expEtag = "foo"
+	fixture.d.etag = fixture.server.expEtag // not modified
+	fixture.server.expCode = 0
+	defer fixture.server.stop()
+
+	resp, err := fixture.d.download(ctx, metrics.New())
+	if err != nil {
+		t.Fatal("Unexpected:", err)
+	}
+	if resp.longPoll != fixture.d.longPollingEnabled {
+		t.Fatalf("Unexpected longPoll modification: %t, %t", fixture.server.longPoll, resp.longPoll)
+	}
+
+	fixture.server.longPoll = false
+
+	resp, err = fixture.d.download(ctx, metrics.New())
+	if err != nil {
+		t.Fatal("Unexpected:", err)
+	}
+	if resp.longPoll != fixture.d.longPollingEnabled {
+		t.Fatalf("Unexpected longPoll modification: %t, %t", fixture.server.longPoll, resp.longPoll)
+	}
+
+}
+
+func TestOneShotLongPollingSwitch(t *testing.T) {
+	ctx := context.Background()
+	config := Config{}
+	timeout := int64(3) // this will result in the test server sleeping for 3 seconds
+	config.Polling.LongPollingTimeoutSeconds = &timeout
+	if err := config.ValidateAndInjectDefaults(); err != nil {
+		t.Fatal(err)
+	}
+	fixture := newTestFixture(t)
+	fixture.d = New(config, fixture.client, "/bundles/test/bundle1").WithCallback(fixture.oneShot)
+	fixture.server.expCode = 0
+	defer fixture.server.stop()
+
+	fixture.server.longPoll = true
+	err := fixture.d.oneShot(ctx)
+	if err != nil {
+		t.Fatal("Unexpected:", err)
+	}
+	if fixture.d.longPollingEnabled != fixture.server.longPoll {
+		t.Fatalf("Unexpected longPollingEnabled %t", fixture.d.longPollingEnabled)
+	}
+
+	fixture.server.longPoll = false
+	err = fixture.d.oneShot(ctx)
+	if err != nil {
+		t.Fatal("Unexpected:", err)
+	}
+	if fixture.d.longPollingEnabled != fixture.server.longPoll {
+		t.Fatalf("Unexpected longPollingEnabled: %t", fixture.d.longPollingEnabled)
+	}
+}
+
+func TestOneShotNotLongPollingSwitch(t *testing.T) {
+	ctx := context.Background()
+	config := Config{}
+	config.Polling.LongPollingTimeoutSeconds = nil
+	if err := config.ValidateAndInjectDefaults(); err != nil {
+		t.Fatal(err)
+	}
+	fixture := newTestFixture(t)
+	fixture.d = New(config, fixture.client, "/bundles/test/bundle1").WithCallback(fixture.oneShot)
+	fixture.server.expCode = 0
+
+	defer fixture.server.stop()
+
+	fixture.server.opaVendorMediaTypeEnabled = true
+	err := fixture.d.oneShot(ctx)
+	if err != nil {
+		t.Fatal("Unexpected:", err)
+	}
+	if fixture.d.longPollingEnabled != true {
+		t.Fatalf("Unexpected longPollingEnabled %t:", fixture.d.longPollingEnabled)
+	}
+
+	fixture.server.opaVendorMediaTypeEnabled = false
+	err = fixture.d.oneShot(ctx)
+	if err != nil {
+		t.Fatal("Unexpected:", err)
+	}
+	if fixture.d.longPollingEnabled {
```

Comment:
Nit: Can we update these error messages to something like

```go
t.Fatal("Expected long polling to be disabled")
```

Reply:
done